Shared/Samples/Create load report/CreateLoadReportView.Model.swift

```diff
@@ -0,0 +1,292 @@
+        private let utilityNetwork = UtilityNetwork(url: .utilityNetwork)
+        
+        /// The initial conditional expression.
+        private var initialExpression: UtilityTraceConditionalExpression?
+        
+        /// The trace parameters for creating load reports.
+        private var traceParameters: UtilityTraceParameters?
+        
+        /// The network attributes for the comparison.
+        private var phasesNetworkAttribute: UtilityNetworkAttribute?
+        
+        /// A list of possible phases populated from the network's attributes.
+        /// By default, they are not included in the load report.
+        @Published private(set) var excludedPhases = [CodedValue]()
+        
+        /// A list of phases that are included in the load report.
+        @Published private(set) var includedPhases = [CodedValue]()
+        
+        /// A list of possible phases populated from the network's attributes.
+        private var allPhases = [CodedValue]()
+        
+        /// The phase summaries in the load report.
+        private var summaries = PhaseSummaries()
+        
+        /// A Boolean value indicating if the run button is enabled.
+        @Published private(set) var runEnabled = false
+        
+        /// The status text to display to the user.
+        @Published private(set) var statusText: String?
+        
+        /// A Boolean value indicating if the sample is authenticated.
+        private var isAuthenticated: Bool {
+            !ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.isEmpty
+        }
+        
+        // MARK: Methods
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await loadUtilityNetwork()
+        }
+        
+        /// Loads the utility network.
+        private func loadUtilityNetwork() async throws {
+            statusText = "Loading utility network…"
+            try await utilityNetwork.load()
+            statusText = nil
+            
+            guard let startingLocation = makeStartingLocation(),
+                  // Get the base condition and trace configuration from a default tier.
+                  let traceConfiguration = getTraceConfiguration() else { return }
+            
+            // Set the default expression.
+            initialExpression = traceConfiguration.traversability?.barriers as? UtilityTraceConditionalExpression
+            
+            // Create downstream trace parameters with elements and function outputs.
+            let traceParameters = UtilityTraceParameters(traceType: .downstream, startingLocations: [startingLocation])
+            traceParameters.addResultTypes([.elements, .functionOutputs])
+            
+            guard let definition = utilityNetwork.definition else { return }
```

Comment:
this function returns early a lot without actually actually failing in any way that the caller of it would know that it actually failed.